bin/pre-commit.sh

```diff
@@ -1,37 +1,23 @@
 #!/bin/bash
 
-if [ -n "$failIt" ]; then
-  echo "Expecting failIt to be empty."
-  exit 1
-fi
+rc=0
 
-wantEmpty=$(gofmt -s -d -l . 2>&1 )
-if [ -n "$wantEmpty" ]; then
-  printf >&2 'gofmt failed for:\n%s\n' "$wantEmpty"
-  failIt=1
-fi
+go_dirs=$(go list -f {{.Dir}} ./...)
 
-wantEmpty=$(goimports -l .)
-if [ -n "$wantEmpty" ]; then
-  printf >&2 'goimports failed for:\n%s\n' "$wantEmpty"
-  failIt=1
-fi
+echo "Running go fmt"
+gofmt -s -d -l ${go_dirs}
+rc=$(($rc || $?))
```

Comment:
Shellcheck tells us that [it's generally better not to include the dollar in a regular variable name in an arithmetic expression](https://github.com/koalaman/shellcheck/wiki/SC2004). So we'd recommend changing `$(($rc || $?))` to `$((rc || $?))`

Reply:
Interesting! I don't think I new I could do that :-)